src/ray/common/id.cc

```diff
@@ -93,48 +220,143 @@ uint64_t MurmurHash64A(const void *key, int len, unsigned int seed) {
   return h;
 }
 
+ActorID ActorID::Of(const JobID &job_id, const TaskID &parent_task_id,
+                    const size_t parent_task_counter) {
+  auto data = GenerateUniqueBytes(job_id, parent_task_id, parent_task_counter,
+                                  ActorID::kUniqueBytesLength);
+  std::copy_n(job_id.Data(), JobID::kLength, std::back_inserter(data));
+  RAY_CHECK(data.size() == kLength);
+  return ActorID::FromBinary(data);
+}
+
+ActorID ActorID::NilFromJob(const JobID &job_id) {
+  std::string data(kUniqueBytesLength, 0);
+  FillNil(&data);
+  std::copy_n(job_id.Data(), JobID::kLength, std::back_inserter(data));
+  RAY_CHECK(data.size() == kLength);
+  return ActorID::FromBinary(data);
+}
+
+JobID ActorID::JobId() const {
+  RAY_CHECK(!IsNil());
+  return JobID::FromBinary(std::string(
+      reinterpret_cast<const char *>(this->Data() + kUniqueBytesLength), JobID::kLength));
+}
+
+TaskID TaskID::ForDriverTask() {
```

Comment:
I think driver task can be `nil task unique bytes + nil actor unique bytes + job id`, instead of random